src/psyclone/psyir/symbols/routinesymbol.py

```diff
@@ -151,6 +152,70 @@ def is_elemental(self, value):
                             f"'{type(value).__name__}'")
         self._is_elemental = value
 
+    def get_routine(self, container=None):
+        '''
+        Recursively searches for the implementation of this RoutineSymbol by
+        following Container imports.
```

Comment:
Thanks Sergi. Just a quick note so that I don't forget: once we support routine interfaces, we will need to support the case where we get more than one possible result for the routine being called (assuming we can't disambiguate using the argument lists).